overlord/snapstate/snapstate_test.go

```diff
@@ -5184,6 +5192,95 @@ func (s *snapmgrTestSuite) TestEnsureRefreshesWithUpdateStoreError(c *C) {
 	c.Check(autoRefreshAssertionsCalled, Equals, 1)
 }
 
+func (s *snapmgrTestSuite) TestEnsureRefreshesDisabledViaSnapdControl(c *C) {
+	st := s.state
+	st.Lock()
+	defer st.Unlock()
+	snapstate.CanAutoRefresh = func(*state.State) (bool, error) { return true, nil }
+
+	makeTestRefreshConfig(st)
+
+	// snapstate.AutoRefresh is called from AutoRefresh()
+	autoRefreshAssertionsCalled := 0
+	restore := mockAutoRefreshAssertions(func(st *state.State, userID int) error {
+		autoRefreshAssertionsCalled++
+		return nil
+	})
+	defer restore()
+
+	// set to managed refresh schedule
+	tr := config.NewTransaction(st)
+	tr.Set("core", "refresh.schedule", "managed")
+	tr.Commit()
+
+	// using MockSnap, we want to read the bits on disk
```

Comment:
This comment is a bit weird. Are we using MockSnap or MockReadInfo?

Reply:
I updated the comment now, there are a bunch of places this is copyied^Wcargo-culted^Wused, if we are happy with the update text I can fix it in all places (probably in a separate PR).